### PR TITLE
shcore-sys: Preliminary support for "High DPI" FFI functions

### PR DESCRIPTION
**Pull request by CruzBishop**
This should be complete, now. Sorry :)

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -301,6 +301,32 @@ pub type size_t = c_uint;
...
+    SHELL_UI_COMPONENT_DESKBAND = 2
+}
+
+#[repr(C)]
+#[derive(Clone, Copy, Debug)]
+pub enum MONITOR_DPI_TYPE {
+    MDT_EFFECTIVE_DPI = 0,
+    MDT_ANGULAR_DPI = 1,
+    MDT_RAW_DPI = 2,
+    //MDT_DEFAULT = MDT_EFFECTIVE_DPI
```
Comment (CruzBishop) [inline src/lib.rs]: Also not sure about this. Is this even allowed in Rust?

Comment (retep998) [inline src/lib.rs]: It's not allowed because it reuses the same discriminant.

Comment (CruzBishop) [inline src/lib.rs]: Alright, thanks :) I've updated the PR with what you suggested

**Comment by retep998** [comment]
Also, just so you know. Your computer's time/timezone is set incorrectly.

**Comment by CruzBishop** [comment]
...Great. Thanks for letting me know <,< I must have messed it up when setting up the Arch/Windows dual boot

**Comment by CruzBishop** [comment]
Just sent up a commit with the real time. Thanks :)

File: src/lib.rs [hunk 2]
```diff
@@ -301,6 +301,33 @@ pub type size_t = c_uint;
 #[cfg(target_arch = "x86_64")]
 pub type size_t = __uint64;
 
+// ShellScalingApi.h
```
Comment (retep998) [inline src/lib.rs]: Split this into `shellscalingapi.rs` please.

File: src/lib.rs [hunk 3]
```diff
@@ -301,6 +301,33 @@ pub type size_t = c_uint;
...
+#[repr(C)]
```
Comment (retep998) [inline src/lib.rs]: Enums should be `#[repr(i32)]`.

Comment (CruzBishop) [inline src/lib.rs]: Oh, okay. I'll fix that up in a minute, thanks :)

**Comment by CruzBishop** [comment]
Okay, now I'm confused. I can't see any reason that it isn't picking up on shellscalingapi.rs, but it won't work :( I'll have a coffee and try again

File: src/lib.rs [hunk 4]
```diff
@@ -53,6 +53,7 @@ pub use propsys::*;
 pub use rpcdce::*;
 pub use schannel::*;
 pub use shellapi::*;
+pub use shellscalingapi::*;
```
Comment (retep998) [inline src/lib.rs]: You forgot the `mod shellscalingapi;` in the mod section.

File: lib/shcore-sys/src/lib.rs [hunk 5]
```diff
@@ -1,9 +1,16 @@
 // Copyright © 2015, Peter Atashian
 // Licensed under the MIT License <LICENSE.md>
 //! FFI bindings to shcore.
-#![no_std]
-#![experimental]
+#![cfg(windows)]
 extern crate winapi;
 use winapi::*;
 extern "system" {
+    //Need to verify return value and HMONITOR handle before enabling
```
Comment (retep998) [inline lib/shcore-sys/src/lib.rs]: You should finish this :stuck_out_tongue:

File: src/lib.rs (hunk 4 above)
Comment (CruzBishop) [inline src/lib.rs]: :dizzy:

Comment (CruzBishop) [inline src/lib.rs]: ...*hides in the corner*

File: lib/shcore-sys/src/lib.rs (hunk 5 above)
Comment (CruzBishop) [inline lib/shcore-sys/src/lib.rs]: I was stuck with HRESULT, but it turns out it was already implemented. Fixing now :)

Comment (CruzBishop) [inline lib/shcore-sys/src/lib.rs]: Ugh, I mean HMONITOR

**Comment by CruzBishop** [comment]
There we go :) Thank you for being so patient.

File: src/shellscalingapi.rs [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+#[repr(i32)]
+#[derive(Clone, Copy, Debug)]
+pub enum MONITOR_DPI_TYPE {
+    MDT_EFFECTIVE_DPI = 0,
+    MDT_ANGULAR_DPI = 1,
+    MDT_RAW_DPI = 2
+}
+
+pub const MDT_DEFAULT: MONITOR_DPI_TYPE = MONITOR_DPI_TYPE::MDT_EFFECTIVE_DPI;
```
Comment (retep998) [inline src/shellscalingapi.rs]: Eof newline missing.

File: src/shellscalingapi.rs [hunk 7]
```diff
@@ -0,0 +1,26 @@
+// ShellScalingApi.h
```
Comment (retep998) [inline src/shellscalingapi.rs]: Missing the license blurb at the top.

File: src/shellscalingapi.rs (hunk 6 above)
Comment (CruzBishop) [inline src/shellscalingapi.rs]: Thanks, fixed

File: src/shellscalingapi.rs (hunk 7 above)
Comment (CruzBishop) [inline src/shellscalingapi.rs]: Oops :cop:

File: lib/shcore-sys/src/lib.rs [hunk 8]
```diff
@@ -1,9 +1,14 @@
 // Copyright © 2015, Peter Atashian
 // Licensed under the MIT License <LICENSE.md>
 //! FFI bindings to shcore.
-#![no_std]
-#![experimental]
+#![cfg(windows)]
 extern crate winapi;
 use winapi::*;
 extern "system" {
+    pub fn GetDpiForMonitor(hmonitor: HMONITOR, dpiType: MONITOR_DPI_TYPE, dpiX: *uint, dpiY: *uint) -> HRESULT
```
Comment (retep998) [inline lib/shcore-sys/src/lib.rs]: `UINT`, not `uint`.

Comment (CruzBishop) [inline lib/shcore-sys/src/lib.rs]: Thanks (again), fixed :)

**Pull request merged**